src/Common/FunctionDocumentation.cpp

```diff
@@ -22,12 +22,121 @@ VersionNumber VERSION_UNKNOWN = {0};
 /// `x` divisible by `y`.
 /// )";
 
+void formatTypes(std::string & result, const std::vector<std::string> & types)
+{
+    /// Generate the argument/return type. To save writing, append links to the documentation on-the-fly
+    ///
+    /// We assume that if 'type' is empty(), 'description' already ends with the elegible type. This check can be removed after all
+    /// existing FunctionDocumentation definitions were changed to provide proper values for the 'type' vector.
+    ///
+    /// Thinking about it, there may be special cases where the exact argument type depends on something else or needs explanation
+    /// otherwise. This cannot be represented using the 'types' vector. Perhaps it is a good idea to leave the check just for that
+    /// reason.
+
+
+    /// Vector 'types' are expected to look like: {"(U)Int*", "Float*"}
+    /// The output is expected to look like: [`(U)Int*`](/sql-reference/data-types/int-uint) or [`Float*`](/sql-reference/data-types/float)
+    bool is_first = true;
+    for (const auto & type : types)
+    {
+        if (is_first)
+            is_first = false;
+        else
+            result += " or "; /// or whatever other separator we want here :-)
+
+        result += "[`" + type;
+
+        if (type.starts_with("String"))
+            result += "`](/sql-reference/data-types/string)";
+        else if (type.starts_with("FixedString"))
+            result += "`](/sql-reference/data-types/fixedstring)";
+        else if (type.starts_with("Int") || type.starts_with("UInt") || type == "(U)Int*") /// matches 'Int8', 'Int16' etc. and 'UInt8', 'UInt16' etc. and 'Int*' and '(U)Int*'
+            result += "`](/sql-reference/data-types/int-uint)";
+        else if (type.starts_with("Float") || type.starts_with("BFloat")) /// matches 'Float32', 'Float64', 'BFloat16' etc
```

Comment:
You can remove `etc` at the end of the line. The enumeration of the three types is exhaustive.